server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -1341,12 +1352,18 @@ private ConnectionStatus getConnectionStatus(DiscoveryNode sender, DiscoveryNode
                 connectionStatus = ConnectionStatus.BLACK_HOLE;
             } else if (disconnectedNodes.contains(sender.getId()) || disconnectedNodes.contains(destination.getId())) {
                 connectionStatus = ConnectionStatus.DISCONNECTED;
-            } else {
+            } else if (nodeExists(sender) && nodeExists(destination)) {
                 connectionStatus = ConnectionStatus.CONNECTED;
+            } else {
+                connectionStatus = ConnectionStatus.DISCONNECTED;
```

Comment:
I think it'd be good to test both `DISCONNECTED` and `BLACK_HOLE` here, perhaps using mostly the same value for the duration of a test.

Reply:
ok